server.go

```diff
@@ -262,14 +222,15 @@ func (s *Server) ListenAndServe() {
 		// connection problems and have call Stop(). If the channel isn't
 		// read/closed within 500ms, retry.
 		if err != nil {
+			s.errorLog("server: got error: %v, will reconnect in %v second(s)", err, 0.5)
```

Comment:
We should probably take the time to define the timeout as a proper duration and use both here when logging and in the `time.After` case to ensure they don't diverge (+ have Go format the time).

Also won't this yield false positives, I think `err` is not nil when `stopChan` is closed (at least according to the code flow?). If no, we should be able to remove the `case _, ok := <-s.stopChan` case and just have the code fall through.

Reply:
I moved it because it was false already since it logged after having waited.

I want to change the logging completely to use the new `slog` package instead. But that is a later PR.

If `stopChan` is closed the log would be:

> server: got error: Blah, will reconnect in 0.5 second(s)
> server: the stopChan was triggered in a reconnect loop, exiting

Which I think it totally fine.